app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java

```diff
@@ -165,78 +166,90 @@ public static boolean isGithubApk(@NonNull final App app) {
     }
 
-    @Nullable
-    public static Disposable checkNewVersion(@NonNull final App app) {
+    private void checkNewVersion() throws IOException, ReCaptchaException {
+        final App app = App.getApp();
+
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(app);
         final NewVersionManager manager = new NewVersionManager();
 
         // Check if user has enabled/disabled update checking
         // and if the current apk is a github one or not.
         if (!prefs.getBoolean(app.getString(R.string.update_app_key), true) || !isGithubApk(app)) {
-            return null;
+            return;
         }
 
         // Check if the last request has happened a certain time ago
         // to reduce the number of API requests.
         final long expiry = prefs.getLong(app.getString(R.string.update_expiry_key), 0);
         if (!manager.isExpired(expiry)) {
-            return null;
+            return;
+        }
+
+        final Response response = getLatestNewPipeVersionsData();
+        handleResponse(response, manager, prefs, app);
+    }
+
+    private Response getLatestNewPipeVersionsData() throws IOException, ReCaptchaException {
+        // Make a network request to get latest NewPipe data.
+        return DownloaderImpl.getInstance().get(NEWPIPE_API_URL);
+    }
+
+    private void handleResponse(@NonNull final Response response,
+                                @NonNull final NewVersionManager manager,
+                                @NonNull final SharedPreferences prefs,
+                                @NonNull final App app) {
+        try {
+            // Store a timestamp which needs to be exceeded,
+            // before a new request to the API is made.
+            final long newExpiry = manager
+                    .coerceExpiry(response.getHeader("expires"));
+            prefs.edit()
+                    .putLong(app.getString(R.string.update_expiry_key), newExpiry)
+                    .apply();
+        } catch (final Exception e) {
+            if (DEBUG) {
+                Log.w(TAG, "Could not extract and save new expiry date", e);
+            }
+        }
+
+        // Parse the json from the response.
+        try {
+            final JsonObject githubStableObject = JsonParser.object()
+                    .from(response.responseBody()).getObject("flavors")
+                    .getObject("github").getObject("stable");
+
+            final String versionName = githubStableObject
+                    .getString("version");
+            final int versionCode = githubStableObject
+                    .getInt("version_code");
+            final String apkLocationUrl = githubStableObject
+                    .getString("apk");
+
+            compareAppVersionAndShowNotification(app, versionName,
+                    apkLocationUrl, versionCode);
+        } catch (final JsonParserException e) {
+            // connectivity problems, do not alarm user and fail silently
```

Comment:
Not a connectivity problem, but most likely something went wrong in https://github.com/teamNewPipe/web-api

Reply:
please update this comment to be correct

Reply:
I'll fix that in a moment!